common/djangoapps/student/models.py

```diff
@@ -1293,6 +1293,32 @@ class Meta:  # pylint: disable=missing-docstring
     def __unicode__(self):
         return "[CourseEnrollmentAllowed] %s: %s (%s)" % (self.email, self.course_id, self.created)
 
+    @classmethod
+    def may_enroll_in(cls, course_id):
+        """
+        Return QuerySet of students who are allowed to enroll in a course.
+
+        Note that result *includes* students who have already enrolled
+        in the course.
+
+        `course_id` identifies the course for which to compute the QuerySet.
+        """
+        return CourseEnrollmentAllowed.objects.filter(course_id=course_id)
+
+    @classmethod
+    def num_may_enroll_in(cls, course_id):
```

Comment:
Given the name of this method and the name of the above method, I would expect this method to return the number of the last method (ie `len(may_enroll_in)`). However that's not the case, which is really confusing. I will read further to try to figure out how you're using these and make suggestions, but this can't stay as-is.